Fix some cards shown twice in SelectCard component

Lynesth opened the pull request:
Interrupts are now processed when they are displayed so there's no need for cards to add themselves to the selection as they are already in playedCards when the interrupt fires.

Fixes #1640

nwai90 commented:
@Lynesth This looks right to me. Do you think we can also remove the code block in `src/interrupts/SelectResourceCard.ts:26`? It seems that this logic is no longer used. We can probably deprecate `optionalCard` at some point as it is no longer necessary to pass in this argument.

```
// Played card is not into playedCards array yet
if (this.optionalCard !== undefined) {
    resourceCards.push(this.optionalCard);
}
```

Lynesth commented:
@nwai90 You are most correct. I removed it.
Fixed double quotes in files I modified and reordered imports in Game.ts since I had to remove the ICard one.

kberg commented:
Looks great. I'm glad this wasn't fixed by removing duplicates from the array.
